### ISSUE TITLE
Page fails to load, api call error in logs

### ISSUE THREAD

**Issue by Asharas** [cem3cem3/CostCalculatorIssues#13]
Hi!  

I just tried the latest version of the mod on a 1 day old server.  
When users try to visit the page it fails to load and the web logs contain the following error:  

```
[2022-04-30 01:21:32.025][Error] Error during execution of /api/v1/costcalculator/stores System.NullReferenceException: Object reference not set to an instance of an object.
   at CostCalculator.CostCalculatorController.getStoreData()
   at lambda_method37183(Closure , Object , Object[] )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
```  

Is this a server or a mod error ?  

Thanks in advance!

**Comment by cem3cem3** [comment]
I have just pushed out a new version of the mod with some exception handling for this issue.
Instead of causing the website to not load, it should now print an error message in the server log and skip the problematic entry.

If you can figure out what causes it specifically, please let me know so I can handle the issue properly.
A quick note: the error message will only be printed once per session, as not to spam the server logs.
If you can provide a savefile, that would highly appreciated too!

**Comment by Asharas** [comment]
Unfortunately it now throws an error in the web browser:  

```
@https://eco.asharas.wtf/costcalculator.wasm:wasm-function[481]:0xda46
@https://eco.asharas.wtf/costcalculator.wasm:wasm-function[682]:0x14dea
@https://eco.asharas.wtf/costcalculator.wasm:wasm-function[1517]:0x3d303
_emscripten_start_fetch/reportSuccess/<@https://eco.asharas.wtf/costcalculator.js line 4 > srcScript:1:65380
callUserCallback@https://eco.asharas.wtf/costcalculator.js line 4 > srcScript:1:56319
reportSuccess@https://eco.asharas.wtf/costcalculator.js line 4 > srcScript:1:65333
fetchXHR/xhr.onload@https://eco.asharas.wtf/costcalculator.js line 4 > srcScript:1:60865
```  

Server log contains an error for each store in the world.  
```
Apr 30 15:17:03 eco EcoServer[14944]: [03:17:03] Cost Calculator error while building reponse for store trade: Boutique.
Apr 30 15:17:03 eco EcoServer[14944]:  Please report this issue to the relevant mod author!
Apr 30 15:17:03 eco EcoServer[14944]: System.Exception: Exception of type 'System.Exception' was thrown.
Apr 30 15:17:03 eco EcoServer[14944]:    at CostCalculator.CostCalculatorController.getStoreData()
Apr 30 15:17:03 eco EcoServer[14944]: [03:17:03] Cost Calculator error while building reponse for store trade: Boutique de Chasse et Charcuterie.
Apr 30 15:17:03 eco EcoServer[14944]:  Please report this issue to the relevant mod author!
Apr 30 15:17:03 eco EcoServer[14944]: System.Exception: Exception of type 'System.Exception' was thrown.
Apr 30 15:17:03 eco EcoServer[14944]:    at CostCalculator.CostCalculatorController.getStoreData()
Apr 30 15:17:03 eco EcoServer[14944]: [03:17:03] Cost Calculator error while building reponse for store trade: Boutique de Compétences.
Apr 30 15:17:03 eco EcoServer[14944]:  Please report this issue to the relevant mod author!
Apr 30 15:17:03 eco EcoServer[14944]: System.Exception: Exception of type 'System.Exception' was thrown.
Apr 30 15:17:03 eco EcoServer[14944]:    at CostCalculator.CostCalculatorController.getStoreData()
Apr 30 15:17:03 eco EcoServer[14944]: [03:17:03] Cost Calculator error while building reponse for store trade: Asha Warma.
Apr 30 15:17:03 eco EcoServer[14944]:  Please report this issue to the relevant mod author!
Apr 30 15:17:03 eco EcoServer[14944]: System.Exception: Exception of type 'System.Exception' was thrown.
Apr 30 15:17:03 eco EcoServer[14944]:    at CostCalculator.CostCalculatorController.getStoreData()
Apr 30 15:17:03 eco EcoServer[14944]: [03:17:03] Cost Calculator error while building reponse for store trade: Boutique de Hunty.
Apr 30 15:17:03 eco EcoServer[14944]:  Please report this issue to the relevant mod author!
Apr 30 15:17:03 eco EcoServer[14944]: System.Exception: Exception of type 'System.Exception' was thrown.
Apr 30 15:17:03 eco EcoServer[14944]:    at CostCalculator.CostCalculatorController.getStoreData()
```  

web logs error is the same as before.  

Here's a link for a recent save of my world (valid 1 month): https://demo.jirafeau.net/f.php?h=2U9bgJtv  

Unfortunately I don't know how to dig deeper in this but I'd be happy to do so if you can provide guidance.

**Comment by cem3cem3** [comment]
Ummm, so I made a mistake in that release. Thank you for replying so fast, and getting me to notice!
I accidentally left some testing code in there that caused an exception to be thrown every time...
Sorry for the inconvenience...
I updated the same version with that code removed. Replace the 2.6.0.9.5.0R you downloaded with the new copy. It should be different.
Thank you for the world download too. Ill take a look at it.

**Comment by Asharas** [comment]
While the server logs don't contain the errors, the one thrown in the web browser still remains, same as pasted in my previous comment.

**Comment by cem3cem3** [comment]
Alright, released version 2.7.0.9.5.0R. Here's hoping I managed to fix all the issues you were getting this time around.

**Comment by Asharas** [comment]
Hey, the page loads know :)  
I haven't tested any further but I'll open another issue if I run into some bug.  

Thanks for the reactivity, I really like the mod and have been eyeballing my prices without it ^^
